branchLayout/api/src/main/java/com/virtuslab/branchlayout/api/BranchLayoutEntry.java

```diff
@@ -16,27 +16,43 @@
  *  Two {@code BranchLayoutEntry} objects are equal when their names, their custom annotations
  *  and their children are ALL equal (recursively checked for children).
  */
-@AllArgsConstructor
+@AllArgsConstructor // for Lombok-generated `withChildren`
 @SuppressWarnings("interning:not.interned") // to allow for `==` comparison in Lombok-generated `withChildren` method
 @ToString
-public class BranchLayoutEntry {
+public final class BranchLayoutEntry {
   @Getter
   private final String name;
 
   @Getter
   private final @Nullable String customAnnotation;
 
+  @Getter
+  private @Nullable BranchLayoutEntry parent = null;
+
   @Getter
   @With
   private final List<BranchLayoutEntry> children;
 
+  public BranchLayoutEntry(String name, @Nullable String customAnnotation, List<BranchLayoutEntry> children) {
+    this.name = name;
+    this.customAnnotation = customAnnotation;
+    this.children = children;
+
+    // Note: since the class is final, `this` is already @Initialized at this point.
+    // We're thus safe to store `this` in other objects' fields.
+
+    for (val child : children) {
+      child.parent = this;
```

Comment:
and it is possible to set the parent even though the child class is final? 

![image](https://user-images.githubusercontent.com/19799111/195596444-c22174ca-dc13-4145-bdb1-2f61208cb1fe.png)

Reply:
Finality of the class shouldn't matter here ;) but generally I've changed this entire structure to immutable now